File: front-end/src/services/studentsService.js
import http from "../http-common";

class studentsService {

    getAllStudents() {
        return http.get("/students/getAll");
    }

    insertStudents(data) {
        return http.post("/students/insert", data);
    }

    updateStudents(id, data) {
        return http.put(`/students/update/${id}`, data);
    }

    deleteStudents(id) {
        return http.delete(`/students/delete/${id}`);
    }

    getStudents(id) {
        return http.get(`/students/getStudents/${id}`);
    }
}


export default new studentsService();

File: back-end/src/main/java/com/example/backend/service/impl/StudentsService.java
package com.example.backend.service.impl;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.backend.model.Students;
import com.example.backend.repository.IStudentsRepository;
import com.example.backend.service.IStudentsService;

@Service
public class StudentsService implements IStudentsService {

    @Autowired
    IStudentsRepository studentsRepository;

    @Override
    public Students insertStudents(Students students) {

        return studentsRepository.insertStudents(students);
    }

    @Override
    public List<Students> getAllStudents() {

        return studentsRepository.getAllStudents();
    }

    @Override
    public Students updateStudents(int id, Students students) {

        return studentsRepository.updateStudents(id, students);
    }

    @Override
    public Students deleteStudents(int id) {

        return studentsRepository.deleteStudents(id);
    }

    @Override
    public Students getStudents(int id) {
        
        return studentsRepository.getStudents(id);
    }

}